virtcontainers/persist/fs/fs.go

```diff
@@ -126,12 +126,25 @@ func (fs *FS) ToDisk(ss persistapi.SandboxState, cs map[string]persistapi.Contai
 		return err
 	}
 
+	var dirCreationErr error
+	var createdDirs []string
+	defer func() {
+		if dirCreationErr != nil && len(createdDirs) > 0 {
+			for _, dir := range createdDirs {
+				// remove the persistFile first
+				cfile := filepath.Join(dir, persistFile)
+				os.Remove(cfile)
```

Comment:
Is it better to merge those two removes into one step of os.RemoveAll ?

Reply:
@lifupan 
Your comment has been addressed.